pytorch_lightning/plugins/training_type/training_type_plugin.py

```diff
@@ -41,6 +57,84 @@ def __init__(self) -> None:
         self._model = None
         self._results: Optional[Union[_EVALUATE_OUTPUT, _PREDICT_OUTPUT]] = None
         self._call_configure_sharded_model_hook = True
+        self._precision_plugin = None
+
+        self.optimizers: List = []
+        self.lr_schedulers: List = []
+        self.optimizer_frequencies: List = []
+
+    @property
+    def precision_plugin(self) -> Optional[PrecisionPlugin]:
+        return self._precision_plugin
+
+    @precision_plugin.setter
+    def precision_plugin(self, args: Dict[str, Any]) -> None:
```

Comment:
tbh, I don't like this. This should live in the acceleator connector. You perform the usual checks there and check if this class implements some other logic, but ideally they'd be determined beforehand (maybe as a staticmethod since you don't seem to really need the sate for selection)